File: 0389-find-the-difference/0389-find-the-difference.cs
using System.Collections.Generic;

public class Solution {
    public char FindTheDifference(string s, string t) {
        Dictionary<char, int> counter = new Dictionary<char, int>();

        foreach (char ch in s) {
            if (counter.ContainsKey(ch)) {
                counter[ch]++;
            } else {
                counter[ch] = 1;
            }
        }

        foreach (char ch in t) {
            if (!counter.ContainsKey(ch) || counter[ch] == 0) {
                return ch;
            }
            counter[ch]--;
        }
        return ' ';
    }
}

File: 0389-find-the-difference/0389-find-the-difference.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public char findTheDifference(String s, String t) {
        Map<Character, Integer> counter = new HashMap<>();

        for (int i = 0; i < s.length(); i++) {
            char ch = s.charAt(i);
            counter.put(ch, counter.getOrDefault(ch, 0) + 1);
        }

        for (int i = 0; i < t.length(); i++) {
            char ch = t.charAt(i);
            if (!counter.containsKey(ch) || counter.get(ch) == 0) {
                return ch;
            }
            counter.put(ch, counter.get(ch) - 1);
        }
        return ' ';
    }
}
